apsuite/optimization/scanning.py

```diff
@@ -8,68 +8,19 @@
 class SimpleScan:
     """."""
 
-    @property
-    def ndim(self):
-        """."""
-        return self._ndim
-
-    @ndim.setter
-    def ndim(self, value):
-        """."""
-        self._ndim = value
-
-    @property
-    def position(self):
-        """."""
-        return self._position
-
-    @position.setter
-    def position(self, value):
-        """."""
-        self._position = value
-
     def __init__(self):
         """."""
-        self._lower_limits = _np.array([])
-        self._upper_limits = _np.array([])
-        self._ndim = 0
         self._position = _np.array([])
         self._delta = _np.array([])
         self._curr_dim = 0
         self._stop = False
-        self._thread = _Thread(target=self._optimize, daemon=True)
-        self.initialization()
-
-    def initialization(self):
-        """."""
-        raise NotImplementedError
-
-    def calc_obj_fun(self):
-        """Return arrays with dimension of search space."""
-        raise NotImplementedError
 
     def set_limits(self, upper=None, lower=None):
         """."""
         self._upper_limits = upper
         self._lower_limits = lower
         self.ndim = len(upper)
```

Comment:
I think this should be an explicit property attached to private ttribute `self._dim`. The class interface is cleaner this way (no need to look specific implementation code to access all class interface)

Reply:
Thanks! This class is broken, I'm fixing it. This parameter is deprecated with the current standard.

Reply:
@xresende, could you take a look again, please.